tests/compiler/std/string.ts

```diff
@@ -770,6 +775,12 @@ assert(dtoa(1.1e+128) == "1.1e+128");
 assert(dtoa(1.1e-64) == "1.1e-64");
 assert(dtoa(0.000035689) == "0.000035689");
 
+
+// concat
+
+assert("Hello ".concat("World") == "Hello World");
+assert("".concat("bar") == "bar");
```

Comment:
Could you add more cases?
```ts
"bar".concat("") == "bar"
"".concat("") == ""
```

Reply:
Done.

BTW, is this intended https://github.com/AssemblyScript/assemblyscript/blob/master/std/assembly/string.ts#L87? 

Why is the null check necessary? Isn't the function signature enough?

Reply:
See this patch: https://github.com/AssemblyScript/assemblyscript/pull/1682
null checks was necessary in the past. But for now it's not necessary